### PR TITLE
OIA-41: Support flow persistence

### PR DESCRIPTION
**Pull request by fooker**
Adds a model interface for flows and a plugin interface to persist such flows.

### CODE REVIEW

**Comment by fooker** [comment]
Based on the work in https://github.com/OpenNMS/opennms/pull/5040 and https://github.com/OpenNMS/opennms/pull/5073

File: api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+import org.opennms.integration.api.v1.annotations.Consumable;
+
+/**
+ * A network flow.
+ *
+ * Represents statistics about the communication between two systems.
+ *
+ * @since 1.0.0
+ */
+@Consumable
```
Comment (j-white) [inline api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java:45]: Can be annotated with `@Model` instead

File: api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import java.util.Optional;
+
+import org.opennms.integration.api.v1.annotations.Consumable;
+
+/**
+ * A network flow.
+ *
+ * Represents statistics about the communication between two systems.
+ *
+ * @since 1.0.0
```
Comment (j-white) [inline api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java:43]:
```suggestion
 * @since 1.1.0
```

**Review by j-white — COMMENTED** [review]
Looks good, we should target the `release-1.x` branch and rev the version to `1.1.0-SNAPSHOT` to take account for the *new* interface.

File: api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java (hunk 1 above)
Comment (fooker) [reply api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java:45]: Can you explain some more? My understanding is, this interface is always implemented by core and never meant to be instantiated by the plugin.

**Comment by fooker** [comment]
> Looks good, we should target the `release-1.x` branch and rev the version to `1.1.0-SNAPSHOT` to take account for the _new_ interface.

Version bumped

**Review by cgorantla — APPROVED** [review]
LGTM!

File: api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java (hunk 1 above)
Comment (j-white) [reply api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java:45]: `Consumable` was normally used for services to be pulled from the registry i.e. https://github.com/OpenNMS/opennms-integration-api/blob/master/api/src/main/java/org/opennms/integration/api/v1/distributed/KeyValueStore.java#L39

`Model` is used to identify object model type interfaces i.e. https://github.com/OpenNMS/opennms-integration-api/blob/master/api/src/main/java/org/opennms/integration/api/v1/model/Node.java#L45

Comment (fooker) [reply api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java]: Got it. Thanks.

Does this imply that there should be a implementation in `commons`? Currently my there is no need to ever instantiate a flow in a plugin.

Comment (j-white) [reply api/src/main/java/org/opennms/integration/api/v1/flows/Flow.java]: `commons` implementation not necessary

**Review by j-white — APPROVED** [review]

**Pull request merged**